src/test/java/be/seeseemelk/mockbukkit/damage/DamageTypeMockTest.java

```diff
@@ -0,0 +1,79 @@
+			"{ 'key': 'minecraft:arrow', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:trident', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:mob_projectile', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:fireworks', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:fireball', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt_on_fire', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:unattributed_fireball', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt_on_fire', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:wither_skull', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:thrown', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:indirect_magic', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.0 }",
+			"{ 'key': 'minecraft:thorns', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:enchant.thorns.hit', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:explosion', 'damageScaling': 'ALWAYS', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:player_explosion', 'damageScaling': 'ALWAYS', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:sonic_boom', 'damageScaling': 'ALWAYS', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.0 }",
+			"{ 'key': 'minecraft:bad_respawn_point', 'damageScaling': 'ALWAYS', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'INTENTIONAL_GAME_DESIGN', 'exhaustion': 0.1 }",
+			"{ 'key': 'minecraft:outside_border', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.0 }",
+			"{ 'key': 'minecraft:generic_kill', 'damageScaling': 'WHEN_CAUSED_BY_LIVING_NON_PLAYER', 'sound': 'minecraft:entity.player.hurt', 'deathMessageType': 'DEFAULT', 'exhaustion': 0.0 }"
```

Comment:
Probably a good idea to add this to a MethodSource instead. I think it would make stuff easier to read. You could even make that methodsource load data from a separate file. I'm not that sure about what is best,  but that's the available options (which includes what you're doing).

Reply:
I agree. I initialy was thinking in using the Junit Argument Provider to load this information from the files and use it as the argument source. But i was not sure if this effort was worthed.
If you agree with it, i can update this PR to include it.

Reply:
This would also allows us to resolve this comment, because it would be resolved as a POJO.
https://github.com/MockBukkit/MockBukkit/pull/973#discussion_r1499368542

Reply:
You're free to decide on what you want to do here. I will approve as long as the comment below is fixed